axiom/nr_errors.c

```diff
@@ -14,37 +14,68 @@
 #include "util_strings.h"
 #include "util_time.h"
 
-nr_error_t* nr_error_create(int priority,
+static nr_error_t* nr_error_create_helper(int priority,
                             const char* message,
                             const char* klass,
+                            const char* error_file,
+                            int error_line,
+                            const char* error_context,
+                            int error_no,
                             const char* stacktrace_json,
                             const char* span_id,
                             nrtime_t when) {
   nr_error_t* error;
 
-  if (0 == message) {
-    return 0;
-  }
-  if (0 == klass) {
-    return 0;
-  }
-  if (0 == stacktrace_json) {
-    return 0;
-  }
-
   error = (nr_error_t*)nr_zalloc(sizeof(nr_error_t));
   error->priority = priority;
   error->when = when;
   error->message = nr_strdup(message);
   error->klass = nr_strdup(klass);
   error->stacktrace_json = nr_strdup(stacktrace_json);
+  if (NULL != error_file && NULL != error_context) {
+    error->error_file = nr_strdup(error_file);
+    error->error_line = error_line;
+    error->error_context = nr_strdup(error_context);
+    error->error_no = error_no;
+  }
 
   if (NULL != span_id) {
     error->span_id = nr_strdup(span_id);
   }
   return error;
 }
 
+nr_error_t* nr_error_create(int priority,
+                            const char* message,
+                            const char* klass,
+                            const char* stacktrace_json,
+                            const char* span_id,
+                            nrtime_t when) {
+  if (0 == message || 0 == klass || 0 == stacktrace_json) {
+    return 0;
+  }
+  return nr_error_create_helper(priority, message, klass, NULL, 0, NULL, 0,
+                         stacktrace_json, span_id, when);
+}
+
+nr_error_t* nr_error_create_additional_params(int priority,
+                                              const char* message,
+                                              const char* klass,
+                                              const char* error_file,
+                                              int error_line,
+                                              const char* error_context,
+                                              int error_no,
+                                              const char* stacktrace_json,
+                                              const char* span_id,
+                                              nrtime_t when) {
+  if (0 == message || 0 == klass || 0 == error_file || 0 == error_context
```

Comment:
Please change usages in this file of `0` to `NULL` if used to test a pointer value.  The usage of `0` is from old code and we do not use that form now.

Reply:
Fixed: [ace780f](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/ace780f45742b8df90d08494c340bda271feb495)